app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -230,4 +236,32 @@ class ConversationViewModel @Inject constructor(
             )
         }
     }
+
+    private suspend fun fromMessageModelToMessageContent(message: Message): MessageContent =
+        when (val content = message.content) {
+            is Asset -> {
+                val assetId = content.value.remoteData.assetId
+                val (imgWidth, imgHeight) = when (val metadata = content.value.metadata) {
+                    is Image -> metadata.width to metadata.height
+                    else -> 0 to 0
+                }
+                val decodedImgDataResult = getMessageAsset(
+                    conversationId = message.conversationId,
+                    messageId = message.id
+                ).run {
+                    when (this) {
+                        is MessageAssetResult.Success -> decodedAsset
+                        else -> null
+                    }
+                }
+                when {
+                    content.value.mimeType.contains("image") -> ImageMessage(decodedImgDataResult, width = imgWidth, height = imgHeight)
```

Comment:
maybe we can have enum for the mimetypes , so it can be reused for other cases later ?